### DISCUSSION TITLE
NDI and Macbook Pro M1 - Not Working...

### DISCUSSION THREAD

**Discussion by TomTom-B** [DistroAV/DistroAV#806] [Q&A - Support]
Hello, 

I use a Macbook Pro M1 (Max) with Mac Os Ventura 13.0.1
I use OBS 28.1.2
I follow all instructions : 
"Download and run obs-ndi-4.10.0-Qt6-macOS.pkg (provided below)
 Install the NDI 4.5.1 macOS runtime using the installer provided here: https://ndi.palakis.fr/runtime/ndi-runtime-4.5.1-macOS.pkg"

But each time i use OBS, it's start with this alert : 
"The NDI™ library cannot be found on your system.
Download the runtime installer here: https://ndi.palakis.fr/runtime/ndi-runtime-4.5.1-macOS.pkg"
In French ;-) 
"Le bibliothèque NDI™ est introuvable sur votre système.
Téléchargez l'installateur du runtime ici : https://ndi.palakis.fr/runtime/ndi-runtime-4.5.1-macOS.pkg"

And after installation, the sound from OBS was stopped and i need to make a new profil... 
Then no NDI but new problems ;-)

Anyone could help me ? 
Thanks. 
Thomas
<img width="424" alt="Capture d’écran 2022-12-09 à 21 47 08" src="https://user-images.githubusercontent.com/120220748/206794030-58f94ff1-b8e2-4bc8-b327-8c6c068302f4.png">

**Comment by nunowonder** [comment]
Hi, you should install obs-ndi-5.0.0-macos-arm64.pkg. Also, install the latest NDI tools.
Anyway let me warn you that the audio right channel of the plugin will be distorting... that's something that will be corrected hopefully in the next release. I know you won't see this version but just search for it on google or something.
Anyway, there's a workaround. You shall use NDI Virtual Input, then capture what you want on that program, and then in OBS just add a source of audio Input Capture, select the NDI Audio, and mute the NDI5 plugin. Hope will work for you, I know its not perfect but it's what we have (by "we", I mean M1 users)

**Reply by reinermatthew** [reply]
Hi @nunowonder,

I'm experiencing the same issue, so I was happy to see your response here. Unfortunately, I'm struggling to find the 5.0 installer anywhere in the repo. Is it not ready yet?

**Reply by nunowonder** [reply]
Hi!
I also can't find it anymore, best thing is to use 4.10, I think there is a 4.10.1 anywhere from November 09, this is REALLY frustrating, I'm always trying new things to not have audio buffering, or audio delay, from this awkward thing of using NDI VIRTUAL INPUT app and OBS to capture it...

**Comment by TychoWerner** [comment]
I've gotten it to work:
From OBS Discord thru !m1ndi i got this result:
1) Close OBS
2) Download and install Newtek NDI®️ Studio Monitor and the OBS NDI®️ plugin
3) Open the Terminal app (you can find it in launchpad, comes pre-installed with every Mac)
4) Paste the following command into the terminal

sudo cp "/Applications/NewTek NDI Video Monitor.app/Contents/Frameworks/libndi.4.dylib" /usr/local/lib/libndi.4.dylib

5) Input your password (note: you will not see the password being put in, but it will be working)
6) Open OBS and check if NDI appears in the Tools menu
---
The terminal command will result in not found, but correct path source is:
/Applications/NDI Video Monitor.app/Contents/Frameworks/libndi_advanced.dylib
and it should still be placed in: /usr/local/lib/libndi.4.dylib

So:
sudo cp "/Applications/NDI Video Monitor.app/Contents/Frameworks/libndi_advanced.dylib" /usr/local/lib/libndi.4.dylib
Should do the trick

**Reply by TychoWerner** [reply]
![image](https://user-images.githubusercontent.com/34608981/212061103-48dc2be3-fcf8-4c93-a08e-5f99db101a13.png)
![image](https://user-images.githubusercontent.com/34608981/212061122-e83a0bba-91b9-4bbf-839c-aa02d8340d0e.png)

**Reply by reinermatthew** [reply]
That worked for me @TychoWerner, thank you!

**Reply by TychoWerner** [reply]
Great to hear, I've not found my solution anywhere else yet so this might be it?
Haven't had time to extensively test it but that's everyone personal job :)

**Reply by reinermatthew** [reply]
When I upgraded this plugin to 4.11.0 and installed the updated NDI 5 runtime, I had to copy libndi_advanced.dylib again, as mentioned above, but this time I had to change the name of the file to libndi.dylib to get the OBS plugin to recognize it.

**Comment by nunowonder** [comment]
This is not the problem, I Have NDI installed and running, the thing is that right channel is distorting via plugin inside OBS, but NDI itself does not, that's why I use NDI VIRTUAL INPUT inside M1Max and then capture it in OBS!

**Reply by thelonewolf** [reply]
How do you get NDI virtual input to work? I have NDI tools with VI installed on both and Intel Mac & an M1max MBP, and event though the ndi video monitor can see the camera, obs doesn't even list the VI as a source. I thought maybe it was just OBS but doesn't show as a source in zoom or google meet either. Is there a permission or something missing to get it to work?
